tlatools/org.lamport.tlatools/src/tlc2/module/TLCExt.java

```diff
@@ -326,15 +328,64 @@ public static synchronized Value tlcModelValue(final Value val) {
 		return ModelValue.add(str.val.toString());
 	}
 	
+	private static final StampedLock tlcEval2Lock = new StampedLock();
```

Comment:
Could you explain why a stamped instead of a ReaderWriteLock?

Reply:
I read about its read locking (by far the most common case here) being faster than ReaderWriterLock, something about recursive locking support / the cache effect of timestamps. But also, it looked interesting and I wanted to try it.

While I don't have a strong opinion, I am now a little curious if the difference is measurable at all between the two locking methods. I have some specs that make extensive use of `TLCCache`, so there is some merit in empirically checking if this matters. I'll reply with results when I get back to this.

Reply:
I checked. In a 5 minute 40 seconds run that calls `TLCCache` constantly, no difference is measurable. I changed it to the simpler lock.